lib/hanami/utils/hash.rb

```diff
@@ -61,11 +61,34 @@ def initialize(hash = {}, &blk)
       #   hash.symbolize!
       #
       #   hash.keys    # => [:a, :b]
+      #   hash.inspect # => { :a => 23, :b => { 'c' => ["x", "y", "z"] } }
+      def symbolize!
+        keys.each do |k|
+          v = delete(k)
+          self[k.to_sym] = v
+        end
+
+        self
+      end
+
+      # Convert in-place all the keys to Symbol instances, nested hashes are converted too.
```

Comment:
For the deep_symbolize! method I was keeping the behaviour that exists currently in master. From what I see it does not symbolize nested array of hashes, but I noticed that there is a PR to support that thought: https://github.com/hanami/utils/pull/161

Should I add that behaviour in this PR?